server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -708,6 +708,8 @@ private DefaultSearchContext createSearchContext(SearchRewriteContext rewriteCon
                 searcher, clusterService, indexService, indexShard, bigArrays, threadPool::relativeTimeInMillis, timeout, fetchPhase);
             success = true;
             return searchContext;
+        } catch (IOException e) {
+            return null;
```

Comment:
this leniency looks dangerous?

Reply:
Should I simply propagate the exception? or any other suggestion?